examples/vision/yolov8.py

```diff
@@ -0,0 +1,628 @@
+mapping is used to encode and decode the class labels during training and inference in
+object detection tasks.
+"""
+
+class_ids = [
+    "car",
+    "pedestrian",
+    "trafficLight",
+    "biker",
+    "truck",
+]
+class_mapping = dict(zip(range(len(class_ids)), class_ids))
+
+# Path to images and annotations
+path_images = "/content/data/images/"
+path_annot = "/content/data/annotations/"
+
+# Get all XML file paths in path_annot and sort them
+xml_files = sorted(
+    [
+        os.path.join(path_annot, file_name)
+        for file_name in os.listdir(path_annot)
+        if file_name.endswith(".xml")
+    ]
+)
+
+# Get all JPEG image file paths in path_images and sort them
+jpg_files = sorted(
+    [
+        os.path.join(path_images, file_name)
+        for file_name in os.listdir(path_images)
+        if file_name.endswith(".jpg")
+    ]
+)
+
+"""
+The function below reads the XML file and finds the image name and path, and then
+iterates over each object in the XML file to extract the bounding box coordinates and
+class labels for each object.
+
+The function returns three values: the image path, a list of bounding boxes (each
+represented as a list of four floats: xmin, ymin, xmax, ymax), and a list of class IDs
+(represented as integers) corresponding to each bounding box. The class IDs are obtained
+by mapping the class labels to integer values using a dictionary called `class_mapping`.
+"""
+
+
+def parse_annotation(xml_file):
+    tree = ET.parse(xml_file)
+    root = tree.getroot()
+
+    image_name = root.find("filename").text
+    image_path = os.path.join(path_images, image_name)
+
+    boxes = []
+    classes = []
+    for obj in root.iter("object"):
+        cls = obj.find("name").text
+        classes.append(cls)
+
+        bbox = obj.find("bndbox")
+        xmin = float(bbox.find("xmin").text)
+        ymin = float(bbox.find("ymin").text)
+        xmax = float(bbox.find("xmax").text)
+        ymax = float(bbox.find("ymax").text)
+        boxes.append([xmin, ymin, xmax, ymax])
+
+    class_ids = [
+        list(class_mapping.keys())[list(class_mapping.values()).index(cls)]
+        for cls in classes
+    ]
+    return image_path, boxes, class_ids
+
+
+image_paths = []
+bbox = []
+classes = []
+for xml_file in tqdm(xml_files):
+    image_path, boxes, class_ids = parse_annotation(xml_file)
+    image_paths.append(image_path)
+    bbox.append(boxes)
+    classes.append(class_ids)
+
+"""
+Here we are using `tf.ragged.constant` to create ragged tensors from the `bbox` and
+`classes` lists. A ragged tensor is a type of tensor that can handle varying lengths of
+data along one or more dimensions. This is useful when dealing with data that has
+variable-length sequences, such as text or time series data.
+
+```python
+classes = [
+    [8, 8, 8, 8, 8],      # 5 classes
+    [12, 14, 14, 14],     # 4 classes
+    [1],                  # 1 class
+    [7, 7],               # 2 classes
+ ...]
```

Comment:
I wonder if keras-cv can handle sample with no-box!
cc @ianstenbit

Reply:
Shouldn't be a problem, since we're padding these raggeds to a dense Tensor anyway